### DISCUSSION TITLE
Improve Product Collection block's No Results view

### DISCUSSION THREAD

**Discussion by shaunandrews** [woocommerce/woocommerce#45048] [WooCommerce, Blocks, and Site Editing]
The "No Results" block currently always appears at the bottom of the Product Collections block. The current default contents are an empty Paragraph block.

<img width="685" alt="CleanShot 2023-08-23 at 12 07 50@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/b1e7c174-2cf7-41a9-b6dc-013f0f856484">

Our first improvement here could be to offer some default content, and introduce patterns as a way to provide options builders and merchants and choose from to further customize.

<img width="1762" alt="CleanShot 2023-08-23 at 12 11 09@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/5a02915d-f447-4652-8631-40689a0a80b4">

I also changed the icon for the block to use the Conditional Content, and updated the description to be a little easier to read.

It's a little strange that this block displays all the time, and can make the editor a little cumbersome to work with — especially if you have multiple Collection blocks on the same page. One possible solution is to only show the No Results block when the parent Collection block (or one of its children) is selected.

Alternatively, we could look at a way to use the List View to toggle different view states of our Collection block.

https://github.com/woocommerce/woocommerce-blocks/assets/191598/46adb42d-f2df-4366-925b-c3f1292a1c99

In the video above, you can see the List View showing the No Results block with a new "hidden" indicator icon. Users could use the icon to toggle visibility. Selecting the No Results block hides the Product Template and Pagination blocks, which also display the hidden indicator.

**Comment by kmanijak** [comment]
> Our first improvement here could be to offer some default content, and introduce patterns as a way to provide options builders and merchants and choose from to further customize.

I really like it. Simple improvement that brings lot of value 👍 

> Alternatively, we could look at a way to use the List View to toggle different view states of our Collection block.

That would be aligned with the experience in Mini Cart or Cart blocks where you can toggle between filled and empty Mini Cart block:

Mini Cart example:

https://github.com/woocommerce/woocommerce-blocks/assets/20098064/7462f861-5c82-4edb-b5aa-98e58baf4e85

> In the video above, you can see the List View showing the No Results block with a new "hidden" indicator icon. Users could use the icon to toggle visibility. Selecting the No Results block hides the Product Template and Pagination blocks, which also display the hidden indicator.

That probably requires a contribution to Gutenberg, but it's a nice improvement and could be used in the mentioned Cart, Mini Cart blocks, but also Query Loop which offers the same behavior of displaying "No results" as Product Collection does at the moment.

**Comment by imanish003** [comment]
Thank you for your detailed insights on the "No Results" block within the Product Collection 🙌🏻

>  introduce patterns as a way to provide options builders and merchants and choose from to further customize.

I like this idea. It resonates well with our planned feature where users will see a dialog from which they can choose collections when the Product Collection block is added. 

> List View showing the No Results block with a new "hidden" indicator icon

I'm with @kmanijak on this. This feature could be an excellent addition not just to our project, but potentially to Gutenberg upstream. Its introduction would likely bring an even more streamlined editing process, allowing users to control visibility in an elegant and efficient way.
